### DISCUSSION TITLE
Use a service to decide which kind of parameter use to link external login to existing account

### DISCUSSION THREAD

**Discussion by PiemP** [OrchardCMS/OrchardCore#16023] [Ideas]
Now OrchardCore link user that use external login, to an existing account, by email in the login information. 

I would like to add a way to use a specific service, for each kind of external login, to retrieve the user to link to the external login. Each service can select the user from the OrchardCore database by a specific data. 

In my case email could be related to different user data and for me could be userful to have a service to select the user in the OrchardCore database by another data (like username).

I hope to be understandable in my explaination. 
Thank you.

PS. I have search about something similar in other discussions or issues but I don't have find something similar... I hope to have use the right search pattern

**Comment by Piedone** [comment]
I don't really understand what your question is.

The external login ID->OC user matching are specific to each provider, and I believe are handled according to the capabilities of the provider (like Microsoft Entra ID). What use cases, specifically, are you looking for?

**Reply by PiemP** [reply]
https://github.com/OrchardCMS/OrchardCore/blob/649b7404d3b1d387be936d2d77430e387c5aee20/src/OrchardCore.Modules/OrchardCore.Users/Controllers/AccountController.cs#L387-L415

In this lines of code OrchardCore search about a user with the same email of the user data retrieved from the external login and after show a view that ask about link this login data to an existing account...I would like to make this operation parametrizable by each login method or, at least, disable this behavior if not needed.

Honestly I'm not sure if this code is more necessary...effectively each login provider can specify it parameter to relate it's login information to an existing account.

**Reply by Piedone** [reply]
I see. Then please open an issue about making this customizable per provider. However, if we want to keep the user store generic with `UserManager<IUser>`, then we can only support user lookups that that service supports: this includes email, name, user ID, and can be a custom string key too with `FindByLoginAsync(string loginProvider, string providerKey)`.
